RELEASE: Treasury Dash + 1.3 Final Final

Zayen-X opened the pull request:
**NEW THINGS IN THE BOX**
- Final 1.3 fix which has the "Redeem all" bonds functionality (#440)
- TREASURY DASHBOARD 🎈 🎉 (#444)
- More love
- 
**Bug Fixes**:
- Add Wallet to treasury (#448)
- Changes address for hades (#452)

ohmygirth committed:
1 3 bonds view (#440)

* added claimbonds component and claim bond rows to bondrow.jsx + initial styles

* synced with alpha styles and cleaned a little

* moved some table styles into global config

* table style tweaks

* fixed cell alignments

* mostly there

* modified state handler and added selector - still needs work

* vesting period added. still wip

* removed some useSelectors that didnt seem necessary (#410)

* claim all and stake works (#413)

* resolved app breaking bug on bondRedeem

* dum warnings about non serializable data disabled

* remove import

* claim all and stake working but needs button to disable while pending

* claim all and stake working

* fixed styles and broken table components

* close

* looks like its working

* fixed pending txn text for buttons

* pending general button txt, disabled claim all on all redeem and redemption through redeem helper contract

Co-authored-by: borbesaur <87902542+borbesaur@users.noreply.github.com>

ohmygirth committed:
added trim and loading to bond claim table plus column alignment tweaks (#443)

* added trim and loading to bond claim table plus column alignment tweaks

* align columns for claim better and nowrap on vesting text

ohmygirth committed:
Treasury dashboard overhaul (#444)

* added chart.jsx, created treasury dashboard page, and added recharts

* added stacked area chart and multi line. still cant get them to render properly

* added barchart and messed with stack styling some more

* quick fixes (#270)

* tweaked sidebar styles

* added timestamp formatting and tweaked scale

* Treasurydash fixes (#272)

* Parse float numbers

* Almost working

* Fix headers and runway

* added additional fnumber formattingg, proper axis labels, and domains

* responsive container tweaks for better chart resolution

* responsive container tweaks for better chart resolution

* Add POL (#281)

* Custom Tooltips (#285)

* recreated tooltip changes off of treasury-dash branch

* removed state from TreasuryDashboard into respective reducers.

* deleted .vscode/settings.json & refactored reducers to be better bundled.

* Card headers (#321)

* fixed OHM Staked chart tooltip.

* formatting headers. icons and runaway available extra data missing.

* info icon added.

* added expand icon.

* added extra runway subHeader info.

* minor refactor , improving responsiveness.

* info tooltips showing with unique message for each.

* tweaking tooltip.

* having trouble setting background color according to selected theme.

* successful implementation. need to adjust card positioning to be absolute.

* info tooltip working perfectly.

* improved Chart header responsiveness.

* minor tweak on import (customtooltip.scss)

* cleanup on chart.scss

* removed console.log in TreasuryDashboard.jsx

* decently positioning modal.

* updated tooltip messages.

* removed local .eslint.js changes

* need to do conditional grid for mobile hero metrics + fix runway line chart styles

* fixed hero metric mobile styling

* Chart messages bulletpoint state refactor (#322)

* corrected minor typo in component prop 'ExpandedChart' alignItem to alignItems.

* moved treasury dashboard state from redux onto treasuryData.js

* added spinner when chart is loading. (#328)

* fixed metric sizing for mobile

* hero metric text adjustments and card sizing tweaks

* chart sizing fixes, responsiveness for expanded and label widths + synced with develop

* fixed supply data in hero

* Treasury Dash Fixes (#364)

* testing out horizontal grid lines in totalValueLocket chart. Conditionally rendering in expandedChart and not in normal view chart.

* edited stroke color. Need to figure how to conditionally change stroke color according to theme.

* added global variable in Chart.jsx for tickCount (repeated value across the code).

* additional ticks added to charts (5 total) and horizontal grid lines to expanded view (with conditional coloring according to chosen theme.

* combined POL into single area with green gradient as per design (need to fix tooltip bulletpoints to conditionally recolor according to chosen theme).

* fixed colors for multi line chart.

* removed local .eslintrc.js changes.

* deleting console.log previously used for debugging.

* fixed chart ticks, normal charts have 3 ticks, and expanded charts have 5.

* fixed hero card for treasury card in wider screens.

* Add info on expanded grpahs

* merged develop and updated hero text + scrollbars

* synced

Co-authored-by: drondin <78686347+drondin@users.noreply.github.com>
Co-authored-by: jc992 <51971598+jc992@users.noreply.github.com>
Co-authored-by: Joao Tavares <jc.992@hotmail.com>
Co-authored-by: Drondin <drondin596@gmail.com>

spaceturtleship committed:
Fix bonding bug where input value is 0 but 'You Will Get' is > 0 (#425)

* Remove default when value is 0 - fixes bug where BondPurchase.jsx displays a value for 'You Will Get' despite input being 0

* Check for null value, assign "0" if so

* Ignore input and display error in case of very small bondQuote value - prevents display of "0.0000 OHM" for small amounts and division underflow for extremely small amounts

* Similar check for LP bonding

Zayen-X committed:
Fixes "add to wallet" bug (#448)

* switched component to URL
* imageURL needs to be absolute because thats what metamask uses

Zayen-X committed:
quickfix: Path Separator for token image.

spaceturtleship committed:
290 changing address for hades (#452)

* Remove check of interestDue; prevents Approve button being replaced with Bond button on bonding modal

* Fix hades address being ignored

* Duplicate MuiDrawer style definition in light.js causing sidebar to display over the top of the blurred background in light mode

* Filter out bonds with no interest due from list of claimable bonds

casuwu committed:
Origin/UI/nav update#401 (#431)

* Adding feedback button on side nav / issue #401

* Adding Canny logo to align with #401 ask

* editing icon

* adding feedback element to side nav

Co-authored-by: Erick Delgado <erickdelgado@protonmail.com>

0xdavinchee committed:
Ts refactor pt0 (#434)

* constants and helpers file extensions changed to ts
- index must be .tsx because it holds react components
* helpers and constants refactored
* index.tsx
* or signer

appleseed-iii commented:
The `Your Bonds` Claim button not working. The main claim buttons do work.
![image](https://user-images.githubusercontent.com/80423742/134197481-233d3dc9-5ad9-4da6-9642-2b640f929a7b.png)

fixed here: https://github.com/OlympusDAO/olympus-frontend/pull/457

appleseed-iii committed:
fix BondRedeem button on ClaimRow (#457)

1. fix BondRedeem button by correcting bond.name lookup
2. remove stale BondRedeemV1.jsx, was no longer used anywhere
3. remove `accountBonds` useEffect from ChooseBond, wasn't doing anything except console.logs
4. add skeleton to vestingPeriod

appleseed-iii committed:
api-key update (#458)

This is temporary, we will get the .env API changes in soon after LUSD launch

appleseed-iii reviewed (approved):
let er rip
